app/pages/onboarding/03-restore-wallet/restore-wallet.tsx

```diff
@@ -84,6 +79,23 @@ export const RestoreWallet: React.FC = () => {
           <ErrorText>{error}</ErrorText>
         </ErrorLabel>
       )}
+      {mnemonicLength === 12 && (
+        <Flex
+          mt="base-loose"
+          px="base"
+          py="base-tight"
+          backgroundColor="#FFEDD6"
+          borderRadius="4px"
+        >
+          <Box mr="tight" mt="1px">
+            <ExclamationMarkCircleIcon size="14px" color="#FE9000" />
+          </Box>
+          <Text textStyle="caption" color="#424248" lineHeight="16px">
+            You're using a 12-word seed phrase. It's recommended not to reuse seed phrases generated
```

Comment:
I don't suspect we want to actually discourage this cross-usage. Instead, we may want to use this opportunity to warn them about how they may be entering their Secret Key for apps (from Connect) but not from a previous token purchase.

E.g. "You've entered a 12-word Secret Key. If you also possess a 24-word Secret Key, you may want to enter that instead since it's more likely to contain any STX balance from a previous purchase."

@ryanarndtcm what do you think of this language? Is there perhaps a better way to frame things without having to mention CoinList specifically or something?

Note as well that we should avoid saying "seed phrase" in favor of "Secret Key" for consistency.

Reply:
I thought the whole point was to discourage cross-usage.

Will adjust seed phrase.

Reply:
See https://github.com/blockstack/stacks-wallet/issues/253#issue-707266979 for the original considerations here re: cross-usage. There's no real harm in using 12-words in the Stacks Wallet, so I'm pretty sure we were never looking to discourage it.

Reply:
That's not how I recall it. 

12-word phrases are fine. But the fact we don't generate them in the wallet inferrs they're being reused. The reuse of keys—especially those used on the hosted authenticator—is what we want to discourage.

Reply:
Perhaps we should touch base on this as a team during mid-sprint planning tomorrow, especially in light of our plans to move away from the hosted authenticator in the browser, which are still getting nailed down.

Reply:
As discussed during the team call yesterday, we should go with language that indicates: 

"It appears you've entered a Secret Key originally created elsewhere than the Stacks Wallet. We recommend you create a new Secret Key with the Stacks Wallet for greater security."